test/io/testwfxwriter.py

```diff
@@ -0,0 +1,168 @@
+            self.assertTrue(maxelectrons - 1 <= nelectrons <= maxelectrons)
+
+    def test_mo_normalization(self):
+        """Check if MO section is printed correctly."""
+        fpath = os.path.join(__datadir__,
+                             "data/GAMESS/basicGAMESS-US2017/C_bigbasis.out")
+        data = cclib.io.ccopen(fpath).parse()
+        wfx = cclib.io.wfxwriter.WFXWriter(data)
+
+        normalized_mocoeffs = wfx._normalized_mocoeffs()
+        if len(data.homos) > 1:
+            self.assertEqual(len(normalized_mocoeffs), wfx._no_electrons())
+        else:
+            self.assertEqual(len(normalized_mocoeffs), wfx._no_of_mos())
+        self.assertEqual(len(normalized_mocoeffs[0]), wfx._no_of_prims())
+
+    def test_mo_normalization_dat(self):
+        """Check if MOs are normalized as expected."""
+        fpath = os.path.join(__datadir__,
+                             "data/GAMESS/basicGAMESS-US2017/dvb_sp.out")
+        data = cclib.io.ccopen(fpath).parse()
+        wfx = cclib.io.wfxwriter.WFXWriter(data)
+
+        normalized_mocoeffs_wfx = wfx._normalized_mocoeffs()
+        datfile = os.path.join(__datadir__,
+                               "data/GAMESS/basicGAMESS-US2017/dvb_sp.dat")
+
+        with open(datfile) as file:
+            content = iter(file.readlines())
+
+        normalized_mocoeffs_dat = []
+
+        line = next(content)
+        while 'TOP OF INPUT FILE FOR BADER' not in line:
+            line = next(content)
+
+        while 'END OF INPUT FILE FOR BADER' not in line:
+            if 'MO' in line and 'OCC NO' in line and 'ORB. ENERGY' in line:
+                line = next(content)
+                mo = []
+                while 'MO' not in line and 'END DATA' not in line:
+                    mo.extend([float(x) for x in line.rsplit()])
+                    line = next(content)
+                normalized_mocoeffs_dat.append(mo)
+            else:
+                line = next(content)
+
+        for mo1, mo2 in zip(normalized_mocoeffs_dat, normalized_mocoeffs_wfx):
+            for coeff1, coeff2 in zip(mo1, mo2):
+                self.assertAlmostEqual(abs(coeff1), abs(coeff2), places=4)
+
+
+    def test_programs(self):
+        """Check other programs against reference data."""
+        ref_file = "data/GAMESS/basicGAMESS-US2017/dvb_sp.out"
+        programs = {
+            'ORCA': "data/ORCA/basicORCA4.0/dvb_sp.out",
+            'NWChem': "data/NWChem/basicNWChem6.5/dvb_sp_hf.out",
+            'Psi': "data/Psi/basicPsi4.0/dvb_sp_rhf.out",
+            'GAMESS-UK': "data/GAMESS-UK/basicGAMESS-UK8.0/dvb_sp_hf.out",
+            'Firefly': "data/GAMESS/basicFirefly8.0/dvb_sp.out",
+            'Molpro': "data/Molpro/basicMolpro2012/dvb_sphf.out"
+        }
+        fpath_ref = os.path.join(__datadir__, ref_file)
+        data_ref = cclib.io.ccopen(fpath_ref).parse()
+        wfx_ref = cclib.io.wfxwriter.WFXWriter(data_ref)
+        norm_mat_ref = wfx_ref._norm_mat()[0]
```

Comment:
I couldn't check `_normalized_mocoeffs` directly, because it also depends upon `ccdata.mocoeffs`. `mocoeffs` are different across programs, for same set of input coordinates and basis set.
Hence, I am checking `_norm_mat` instead. It ensures that the other three data used in normalization i.e,  `prim_type`, `alpha`, `prim_coeff` are being correctly computed.